projects/igniteui-angular/src/lib/radio/themes/_base.scss

```diff
@@ -0,0 +1,431 @@
+@use 'sass:map';
+@use 'igniteui-theming/sass/animations' as *;
+@use 'igniteui-theming/sass/bem' as *;
+@use 'igniteui-theming/sass/themes' as *;
+@use 'igniteui-theming/sass/typography' as *;
+@use 'igniteui-theming/sass/color/functions' as *;
+@use './light/tokens' as *;
+
+$_theme: $material;
+
+@layer base {
+    @include scale-in-out($start-scale: .9);
+
+    %radio-label {
+        color: var-get($_theme, 'label-color');
+        user-select: none;
+        word-wrap: break-all;
+    
+        &:empty {
+            display: none;
+        }
+    }
+
+    %radio-ripple--hover {
+        background: var-get($_theme, 'empty-color');
+        transition: background .2s $ease-out-quad;
+        opacity: .06;
+    
+        @container style(--ig-theme-variant: dark) {
+            opacity: .12;
+        }
+    }
+    
+    %radio-ripple--hover-checked {
+        background: var-get($_theme, 'fill-color');
+    }
+    
+    %radio-ripple--hover-invalid {
+        background: var-get($_theme, 'error-color');
+    }
+    
+    %radio-ripple--focused {
+        background: var-get($_theme, 'empty-color');
+        transition: background .2s $ease-out-quad;
+        opacity: .12;
+    
+        @container style(--ig-theme-variant: dark) {
+            opacity: .24;
+        }
+    }
+
+    %radio-ripple--pressed {
+        opacity: .12;
+    
+        @container style(--ig-theme-variant: dark) {
+            opacity: .24;
+        }
+    }
```

Comment:
These lines can be improved to reduce duplication and improve maintainability. See checkbox base for reference.